docs/design/connection-tester.md

```diff
@@ -0,0 +1,50 @@
+# Connection Tester
+
+The goal of this document is to sketch out possible approaches for implementing _Connection testing_ in the context of iPaaS. This backend concept is introduced to support the use case to quickly test whether a given `Connection`  (which is the combination of a `Connector` and a `ConnectorConfiguration`) is actually usable. The typical use case is, that the user pushes a button "Test connection" on the "Connection Details" page which actually triggers the test of the connection and that can be either successful or failing. When it fails, a detailed error message describes not only the error itself but also how to resolve the error (e.g. "Cannot connect because of invalid credentials"). 
+
+## Communication flow
+
+When the user triggers this test function from within the UI, a backend call to `ipaas-rest` will be issued, which in turn will trigger a (still to be defined) method on the referenced connector with the concrete configuration given. This method, let's call it `ping()`, must be side effect free, which should be possible for most endpoints. The method takes the configuration as argument for the connector and returns a result objects which contains the result of the check (ok / nok) and in case of a failure the error message (question: Should we already think about I18N here ?)
```

Comment:
@rhuss I agree with the two types of tests, I just think the Citizen Developer will need this functionality now. We have no control over the 'twitter' endpoint; it can be intermittently down, it can have a bug, it can change over time, and potentially break the current integrations, etc, etc.

Also I think we do know the API contract at any point in the integration to allow for a data mapping step.